include/deal.II/fe/fe.h

```diff
@@ -1308,6 +1308,16 @@ class FiniteElement : public Subscriptor, public FiniteElementData<dim>
   get_subface_interpolation_matrix(const FiniteElement<dim, spacedim> &source,
                                    const unsigned int                  subface,
                                    FullMatrix<double> &matrix) const;
+
+
+  /**
+   * Same function as above with `std::complex<double>`.
+   */
+  virtual void
+  get_subface_interpolation_matrix(
+    const FiniteElement<dim, spacedim> &source,
+    const unsigned int                  subface,
+    FullMatrix<std::complex<double>> &  matrix) const;
```

Comment:
In the file `dofs_tools_constrains.cc` line 1861, I need the `std::complex` version of
```c++
fe.get_subface_interpolation_matrix(fe, c, subface_interpolation);
```
But `get_subface_interpolation_matrix` is virtual, therefore I can not template it. What would you suggest? I did a proof of concept for the class `FE_Q_Base`, see below. Do you have a better solution?